mon-entreprise/source/pages/Simulateurs/IndépendantSimulation.tsx

```diff
@@ -1,19 +1,128 @@
+import PeriodSwitch from 'Components/PeriodSwitch'
 import SimulateurWarning from 'Components/SimulateurWarning'
 import Simulation from 'Components/Simulation'
 import IndépendantExplanation from 'Components/simulationExplanation/IndépendantExplanation'
-export default function IndépendantSimulation() {
+import { SimulationGoal, SimulationGoals } from 'Components/SimulationGoals'
+import { useEngine } from 'Components/utils/EngineContext'
+import { DottedName } from 'modele-social'
+import { useDispatch } from 'react-redux'
+
+export function IndépendantPLSimulation() {
 	return (
 		<>
-			<SimulateurWarning simulateur="indépendant" />
-			<Simulation explanations={<IndépendantExplanation />} />
+			<SimulateurWarning simulateur="profession-libérale" />
+			<Simulation explanations={<IndépendantExplanation />}>
+				<PeriodSwitch />
+				<IndépendantSimulationGoals />
+			</Simulation>
 		</>
 	)
 }
-export function IndépendantPLSimulation() {
+
+export default function IndépendantSimulation() {
 	return (
 		<>
-			<SimulateurWarning simulateur="profession-libérale" />
-			<Simulation explanations={<IndépendantExplanation />} />
+			<SimulateurWarning simulateur="indépendant" />
+			<Simulation explanations={<IndépendantExplanation />}>
+				<div
+					css={`
+						display: flex;
+						flex-wrap: wrap-reverse;
+						> * {
+							margin-top: 0.6rem;
+						}
+						justify-content: center;
+
+						@media (min-width: 590px) {
+							justify-content: space-between;
+						}
+					`}
+				>
+					<ImpositionSwitch />
+					<PeriodSwitch />
+				</div>
+				<IndépendantSimulationGoals />
+			</Simulation>
 		</>
 	)
 }
+function IndépendantSimulationGoals() {
+	return (
+		<SimulationGoals className="plain">
+			<Condition expression="entreprise . imposition = 'IR'">
+				<SimulationGoal
+					appear={false}
+					alwaysShow
+					dottedName="entreprise . chiffre d'affaires"
+				/>
+				<Condition expression="entreprise . imposition . IR . micro-fiscal != oui">
+					<SimulationGoal
+						small
+						appear={false}
+						dottedName="entreprise . charges"
+					/>
+				</Condition>
+			</Condition>
+			<Condition expression="entreprise . imposition = 'IS'">
+				<SimulationGoal
+					appear={false}
+					dottedName="dirigeant . rémunération . totale"
+				/>
+			</Condition>
+
+			<SimulationGoal
+				small
+				editable={false}
+				alwaysShow
+				dottedName="dirigeant . indépendant . cotisations et contributions"
+			/>
+			<Condition expression="entreprise . imposition . IR . micro-fiscal">
+				<SimulationGoal
+					small
+					appear={false}
+					dottedName="entreprise . charges"
+				/>
+			</Condition>
+			<SimulationGoal dottedName="dirigeant . rémunération . nette" />
+			<Condition expression="impôt > 0">
+				<SimulationGoal small editable={false} dottedName="impôt" />
+			</Condition>
+			<SimulationGoal dottedName="dirigeant . rémunération . nette après impôt" />
+		</SimulationGoals>
+	)
+}
+
+function ImpositionSwitch() {
```

Comment:
Je trouve que l'affichage en tant que Switch en haut du simulateur complexifie l'UI, pour un paramètre qui n'est pas vraiment un switch (l'utilisateur ne va pas alterner plusieurs fois entre les 2 modes d'impositions, contrairement à la vue mois/année) mais plutôt un “pré-requis” à la simulation.

Limite je verrais plus un bandeau avec une ligne de texte comme :

« Ce simulateur s’adresse aux sociétés à l'impôt sur le revenu. [Ma société est à l'impôt sur les société] » (avec un lien qui fait la même fonction que le switch).

Évidemment pas bloquant pour merger, je partage juste l'idée.

Reply:
Bonne idée, je suis d'accord. Je pense que ce simulateur en l'état est voué à disparaître, au profit de simulateurs axés type d'entreprise (EURL, EURL à l'IR, EI, EIRL, SASU à l'IS, etc), avec les liens vers la version moins commune (par exemple eirl à l'is) dans la description du haut.